pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -454,6 +456,52 @@ func buildGatewayListenerTLSContext(server *networking.Server, enableSds bool) *
 				},
 			}
 		}
+	} else if server.Tls.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
+		// global SDS must be enabled if the sds path is non-default, configure TLS with SDS
+		if sdsPath != "" {
+			// configure egress with SDS
+			tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_CombinedValidationContext{
+				CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
+					DefaultValidationContext: &auth.CertificateValidationContext{VerifySubjectAltName: server.Tls.SubjectAltNames},
+					ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(
+						authn_model.SDSRootResourceName, sdsPath, metadata),
+				},
+			}
+			tls.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{
+				authn_model.ConstructSdsSecretConfig(
+					authn_model.SDSDefaultResourceName, sdsPath, metadata)}
+		} else {
+			// global SDS disabled, fall back on using mounted certificates
+			caCertificates := model.GetOrDefaultFromMap(nil, model.NodeMetadataTLSClientRootCert, constants.DefaultRootCert)
+			trustedCa := &core.DataSource{
+				Specifier: &core.DataSource_Filename{
+					Filename: caCertificates,
+				},
+			}
+			tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContext{
+				ValidationContext: &auth.CertificateValidationContext{
+					TrustedCa:            trustedCa,
+					VerifySubjectAltName: server.Tls.SubjectAltNames,
+				},
+			}
+
+			certChainPath := model.GetOrDefaultFromMap(nil, model.NodeMetadataTLSClientCertChain, constants.DefaultCertChain)
```

Comment:
NodeMetadataTLSClientCertChain and NodeMetadataTLSClientKey do not sound correct. For a listener at gateway, it is acting as server side. These fields should be NodeMetadataTLSServerCertChain and NodeMetadataTLSServerKey. Correct me if I am wrong. Is there any test to guard this change?

We need to update doc in istio.io to introduce this use case.

Reply:
You are correct, thanks for the catch. Test didn't guard this because it's defaulting to the default mount path (and the defaults are correct). Also, looking again, I think the first parameter here should be `metadata` rather than `nil`.

Definitely all of these changes will need to be included in Traffic Management docs on istio.io